## Kaggle Sentiment Analysis for financial news 데이터 셋 load

https://www.kaggle.com/ankurzing/sentiment-analysis-for-financial-news

In [1]:
path = "./archive/all-data.csv"

In [2]:
all_lines = []

In [3]:
import csv

f = open(path, 'r', encoding="ISO-8859-1")
lines = csv.reader(f)


for line in lines:
    print(line)    
    print('\n')
    all_lines.append(line)    

['neutral', 'According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .']


['neutral', 'Technopolis plans to develop in stages an area of no less than 100,000 square meters in order to host companies working in computer technologies and telecommunications , the statement said .']


['negative', 'The international electronic industry company Elcoteq has laid off tens of employees from its Tallinn facility ; contrary to earlier layoffs the company contracted the ranks of its office workers , the daily Postimees reported .']


['positive', 'With the new production plant the company would increase its capacity to meet the expected increase in demand and would improve the use of raw materials and therefore increase the production profitability .']


['positive', "According to the company 's updated strategy for the years 2009-2012 , Basware targets a long-term net sales growth in the range of 20 % -40 % with an operating 



['positive', 'Kalmar has been awarded a new 5-year contract to supply its Rough Terrain Container Handler RTCH .']


['positive', 'Major Order in India Comptel Corporation has received a significant long-term order for mediation and provisioning solutions being used by a leading operator in India .']


['positive', "Managing Director Kari Inkinen says that Sponda 's leasing operations developed highly favourably ."]


['positive', 'Net cash flow from operations is expected to remain positive .']


['positive', "Of the price , Kesko 's share is 10 mln euro $ 15.5 mln and it will recognize a gain of 4.0 mln euro $ 6.2 mln on the disposal which will be included in the result for the second quarter of 2008 ."]


['positive', 'Olvi has posted a strong set of figures for the first six months of this year .']


['positive', 'Operating profit improved by 27 % to EUR 579.8 mn from EUR 457.2 mn in 2006 .']


['positive', 'Our standardised services have met with a positive reception among our c


['neutral', 'The trucks feature an Eco Drive system - a fuel measuring tool which stores data particular to individual drivers .']


['neutral', 'The veneer and plywood industry in Sweden is quite small despite the raw material resources .']


['neutral', 'The writing and publication of Lemmink+Æinen -\x93 s history is a continuation of earlier collaboration between Antti Tuuri and the company .']


['neutral', 'These products include Personal Communications products such as mobile phones and their parts , Home Communications products such as set-top boxes and electronics for flat panel TVs as well as Communications Networks products such as base-stations , tower-top amplifiers , and microwave systems .']


['neutral', 'They re in a race with Google to get lots of users onto their service as social networking creates new business models , said Martin Garner , a London-based analyst with CCS Insight .']


['neutral', 'This beauty is an updated version of her chichi loop shade launched 

['neutral', "Kemira will supply the additional volumes of iron coagulants from the company 's plant at Helsingborg , Sweden ."]


['positive', "Last month , Outokumpu sold more than 80 percent of its technology unit , Outokumpu Technology Oyj , to further its strategic goal of becoming the world 's largest stainless steel maker ."]


['positive', "Last week , the Finnish metals and technology group announced plans to sell more than 80 percent of its technology unit to further the company 's strategic goal of becoming the world 's largest stainless steel maker ."]


['positive', 'MADISON , Wis. , Feb. 6 - PRNewswire - -- Fiskars announces that the United States Patent and Trademark Office has granted a trademark registration for the color orange on its iconic scissor handles .']


['positive', 'MANAVIGATOR-September 7 , 2010-Kemira unveils Indian JV with IVRCL Finnish chemicals group Kemira ( HEL : KRA1V ) on Tuesday announced it has inked a deal to form a joint venture in India with lo



['neutral', "Danske Bank is Denmark 's largest bank with 3.5 million customers ."]


['neutral', 'Decisions are to be made as quickly as possible .']


['neutral', 'Deliveries by Outotec will take place over 30 months .']


['neutral', 'Deliveries of Nokia 1112 , Nokia 2310 and Nokia 2610 are expected to start in the second quarter of 2006 .']


['neutral', 'Delivery is due in the second half of 2011 .']


['neutral', "Den Bosch-based TomTom is Europe 's largest maker of automotive navigation devices , while Cayman Islands-based Garmin is larger in the U.S. and overall ."]


['neutral', 'Depending on the market situation , such projects are sold after 1 to 3 years after completion .']


['positive', 'Dubbed Nokia Money Solutions , company officials say the initiative would seek to bridge the gap between the banked and unbanked in developing countries .']


['negative', 'Due to market conditions , the company will also temporarily lay off about 3,200 people at different sites .']


['

['neutral', 'In Sweden the agreement covers data communication services as well .']


['neutral', 'In the automobile space , Maruti is the most searched brand in cars .']


['positive', 'In the autumn , it plans to expand service to St. Petersburg and Moscow and launch service to Yekaterinburg , Russia , flying there three times a week .']


['neutral', 'In the beginning of the negotiations , the estimated number of reductions was 50 at the maximum .']


['neutral', "In the Czech Republic , the profiling unit at Ostrava will be closed and the machinery will be relocated to Ruukki 's bigger plants in Hungary , Poland and Romania by the end of the first quarter of 2009 ."]


['neutral', "In the Czech Republic , the smaller profiling unit at Ostrava will be closed and the machinery will be gradually relocated to Ruukki 's bigger plants in Hungary , Poland and Romania by the end of the first quarter of 2009 ."]


['neutral', "In the first quarter of 2008 , Sacanfil 's net sales totalled EU



['negative', 'And U.S. energy executives say high steel prices are threatening energy exploration .']


['negative', 'As capacity was cut with 1.4 % , the passenger load factor was down 7.8 percentage points .']


['negative', 'At present , the trade mark Pit-Produkt is little-known outside the North-West of Russia .']


['negative', 'Copper , lead and nickel also dropped ... HBOS ( HBOS ) plummeted 20 % to 70.3 pence after saying this year+\x88 ??']


['negative', "Divisional sales demand in the half remained `` inconsistent '' , it added ."]


['negative', 'During the strike , Finnair estimates to incur a net loss of between EUR2m and EUR2 .5 m per day .']


['negative', 'Earnings per share ( EPS ) dropped to EUR 0.21 from EUR 0.31 .']


['negative', 'EMSA Deputy Chairman of the Board Juri Lember told BNS on Wednesday that this was the first time he heard about the strike as the Swedish side had not informed the Estonian union yet .']


['negative', "Finnair 's passenger load facto

In [4]:
#데이터 셋의 문장 개수
len(all_lines)

4846

In [5]:
all_lines[0]

['neutral',
 'According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .']

In [6]:
#데이터셋은 neutral, negative, postive 으로 labelling 되어있음
all_lines[0][0]

'neutral'

In [7]:
# 데이터셋의 문장 부분
all_lines[0][1]

'According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .'

## 네이버 파파고 API활용 번역
4846개 문장을 번역한 후 제대로 번역되지 않은 문장은 직접 수정

In [8]:
import os
import sys
import urllib.request
import json

In [9]:
#client_id와 client_secret 정보
client_id = "-"
client_secret = "-"

In [11]:
def translator(text, source='en', target='ko'):  
    
    Text = urllib.parse.quote(text)    
    data = f'source={source}&target={target}&text={Text}'
    
    url = "https://naveropenapi.apigw.ntruss.com/nmt/v1/translation"
    request = urllib.request.Request(url)
    request.add_header("X-NCP-APIGW-API-KEY-ID",client_id)
    request.add_header("X-NCP-APIGW-API-KEY",client_secret)
    response = urllib.request.urlopen(request, data=data.encode("utf-8"))
    rescode = response.getcode()
    if(rescode==200):
        response_body = response.read()
        response = json.loads(response_body.decode('utf-8'))    
        result = response['message']['result']['translatedText']
        return(result)
    else:
        return ("Error")       
        

In [16]:
# #API 테스트 문장
test = "the broker gave an `` outperform '' recommendation on the stock ."
translator(test)

'주식 중개인은 주식에 대해 "상당히" 추천했다.'

In [17]:
done = []

In [18]:
for i in range(len(all_lines)):     
    input = all_lines[i][1]
    result = translator(input)
    done.append(result)   

In [19]:
done

['Gran에 따르면, 비록 그 회사가 성장하고 있는 곳이지만, 그 회사는 러시아에 모든 생산을 옮기는 계획을 가지고 있지 않습니다.',
 '테크노폴리스는 컴퓨터 기술과 통신 분야에서 일하는 회사들을 유치하기 위해 100,000 평방미터 이상의 면적을 단계적으로 개발할 계획이라고 성명은 말했다.',
 '국제적인 전자 산업 회사 Elcoteq는 그것의 Talinn 시설로부터 10명의 직원들을 해고하였습니다; 회사가 그것의 사무직 근로자들의 순위를 수축시킨 것에 반대합니다, 일일 Postimees는 보고하였습니다.',
 '새로운 생산 공장과 함께 회사는 수요의 예상되는 증가를 충족시키기 위하여 그것의 능력을 증가시킬 것입니다 그리고 원자재의 사용을 개선할 것이고 따라서 생산 수익성을 증가시킬 것입니다.',
 '2009-2012년 회사의 최신 전략에 따르면, Basware는 순매출의 10%-20%의 영업이익률로 20%-40%의 장기 순매출 성장을 목표로 하고 있습니다.',
 'ASPOCOMP의 파이낸싱은 기술적으로 더 까다로운 HDI 인쇄 회로 기판 PCB에 점점 더 초점을 맞춤으로써 성장 전략을 적극적으로 추진하고 있습니다.',
 '2010년의 마지막 분기에 대하여, 그것이 EUR7m의 세금 전 손실로부터 0 세전 이익으로 이동한 동안, Componenta의 순매출은 1년 전에 같은 기간 동안 EUR76m에서 EUR131m로 두배가 되었다.',
 '2010년의 3/4분기에, 순매출은 EUR 205.5 mn에 5.2 퍼센트 증가하였습니다, 그리고 영업 이익은 EUR 23.5 mn에 34.9 퍼센트 증가하였습니다.',
 '영업 이익은 순매출의 7.7 퍼센트를 나타내는 2007년 해당 기간에 EUR 8.7 mn에서 13.1 mn으로 증가하였습니다.',
 '영업 이익은 총 21.1 mn이었습니다, 2007년에 EUR 18.6 mn으로부터 위로, 순매출의 9.7 퍼센트를 대표하였습니다.',
 '텔리아소네라 TLSN은 이 제안이 핵심 사업 보유에 대한 소유권을 늘리기 위한 

In [23]:
len(done)

4846

In [25]:
for i in range(len(done)):
    temp = done[i]
    all_lines[i].append(temp)

In [26]:
all_lines

[['neutral',
  'According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .',
  'Gran에 따르면, 비록 그 회사가 성장하고 있는 곳이지만, 그 회사는 러시아에 모든 생산을 옮기는 계획을 가지고 있지 않습니다.'],
 ['neutral',
  'Technopolis plans to develop in stages an area of no less than 100,000 square meters in order to host companies working in computer technologies and telecommunications , the statement said .',
  '테크노폴리스는 컴퓨터 기술과 통신 분야에서 일하는 회사들을 유치하기 위해 100,000 평방미터 이상의 면적을 단계적으로 개발할 계획이라고 성명은 말했다.'],
 ['negative',
  'The international electronic industry company Elcoteq has laid off tens of employees from its Tallinn facility ; contrary to earlier layoffs the company contracted the ranks of its office workers , the daily Postimees reported .',
  '국제적인 전자 산업 회사 Elcoteq는 그것의 Talinn 시설로부터 10명의 직원들을 해고하였습니다; 회사가 그것의 사무직 근로자들의 순위를 수축시킨 것에 반대합니다, 일일 Postimees는 보고하였습니다.'],
 ['positive',
  'With the new production plant the company would increase its capacity to meet the e

In [27]:
len(all_lines)

4846

## 엑셀로 내보내기
리스트 형식의 데이터를 엑셀로 내보냈습니다. 

In [28]:
from openpyxl import Workbook

In [31]:
wb = Workbook()

In [34]:
sheet = wb.active

In [35]:
for i in range(len(all_lines)):
    sheet.append(all_lines[i])

In [36]:
wb.save('Kaggle_translated.xlsx')